sklearn/metrics/_pairwise_distances_reduction/_pairwise_distances.pyx.tp

```diff
@@ -0,0 +1,373 @@
+cdef class EuclideanPairwiseDistances{{name_suffix}}(PairwiseDistances{{name_suffix}}):
+    """EuclideanDistance-specialized {{name_suffix}}bit implementation for PairwiseDistances."""
+
+    @classmethod
+    def is_usable_for(cls, X, Y, metric) -> bool:
+        return (PairwiseDistances{{name_suffix}}.is_usable_for(X, Y, metric)
+                and not _in_unstable_openblas_configuration())
+
+    def __init__(
+        self,
+        X,
+        Y,
+        bint use_squared_distances=False,
+        chunk_size=None,
+        strategy=None,
+        metric_kwargs=None,
+    ):
+        if (
+            metric_kwargs is not None and
+            len(metric_kwargs) > 0 and
+            "Y_norm_squared" not in metric_kwargs
+        ):
+            warnings.warn(
+                f"Some metric_kwargs have been passed ({metric_kwargs}) but aren't "
+                f"usable for this case (EuclideanPairwiseDistances{{name_suffix}}) and will be ignored.",
+                UserWarning,
+                stacklevel=3,
+            )
+
+        super().__init__(
+            # The datasets pair here is used for exact distances computations
+            datasets_pair=DatasetsPair{{name_suffix}}.get_for(X, Y, metric="euclidean"),
+            chunk_size=chunk_size,
+            strategy=strategy,
+        )
+        # X and Y are checked by the DatasetsPair{{name_suffix}} implemented as
+        # a DenseDenseDatasetsPair{{name_suffix}}
+        cdef:
+            DenseDenseDatasetsPair{{name_suffix}} datasets_pair = (
+                <DenseDenseDatasetsPair{{name_suffix}}> self.datasets_pair
+            )
+            ITYPE_t dist_middle_terms_chunks_size = (
+                self.Y_n_samples_chunk * self.X_n_samples_chunk
+            )
+
+        self.gemm_term_computer = GEMMTermComputer{{name_suffix}}(
+            datasets_pair.X,
+            datasets_pair.Y,
+            self.effective_n_threads,
+            self.chunks_n_threads,
+            dist_middle_terms_chunks_size,
+            n_features=datasets_pair.X.shape[1],
+            chunk_size=self.chunk_size,
+        )
+
+        if metric_kwargs is not None and metric_kwargs.get("Y_norm_squared", None) is not None:
+            self.Y_norm_squared = metric_kwargs.pop("Y_norm_squared")
+        else:
+            self.Y_norm_squared = _sqeuclidean_row_norms{{name_suffix}}(
+                datasets_pair.Y,
+                self.effective_n_threads,
+            )
+
+        if metric_kwargs is not None and metric_kwargs.get("X_norm_squared", None) is not None:
+            self.X_norm_squared = metric_kwargs.pop("X_norm_squared")
+        else:
+            # Do not recompute norms if datasets are identical.
+            self.X_norm_squared = (
+                self.Y_norm_squared if self.datasets_pair.X_is_Y else
+                _sqeuclidean_row_norms{{name_suffix}}(
+                    datasets_pair.X,
+                    self.effective_n_threads,
+                )
+            )
```

Comment:
Note: this is safer than what the equivalent lines in other `Euclidean*` specialisation.

Reply:
Can you explain more, I did not understand this note

Reply:
This unpacking in `EuclideanPairwiseDistances` for `{X,Y}_norm_squared` is safer than the one in `EuclideanArgKmin` and `EuclideanRadiusNeighbors`: this one checks for `None` better, where the others do not.

Reply:
> So you mean we should update the order reducing variants to follow the same strategy in a follow-up PR?

Yes, so as to prevent issues like the current ones on the CI as you mentioned in https://github.com/scikit-learn/scikit-learn/pull/23958#issuecomment-1260531381